shared/src/main/scala/amf/core/parser/Reference.scala

```diff
@@ -22,31 +23,35 @@ case class Reference(url: String, refs: Seq[RefContainer]) extends PlatformSecre
     copy(refs = refs :+ RefContainer(kind, ast, fragment))
   }
 
-  def resolve(compilerContext: CompilerContext, allowRecursiveRefs: Boolean)(
+  def resolve(compilerContext: CompilerContext, domainPlugin: AMFParsePlugin, allowRecursiveRefs: Boolean)(
```

Comment:
Passing the list of allowed media types as parameter seems clearer to me. Passing a parse plugin makes me assume that plugin will parse the reference.

Reply:
Agreed.